adds optional menu to rename workspaces

rk1a opened the pull request:
Hi, I added a simple menu to directly rename workspaces. 
![image](https://user-images.githubusercontent.com/96744960/147767446-27aaf3be-be39-4728-b29c-a018cabe8c2f.png)
When it is enabled in the preferences, a right click on the workspace bar opens the menu, a left click switches to the clicked workspace as before.

rk1a committed:
adds optional menu to rename workspaces

rk1a committed:
improves interaction with popup menu

rk1a committed:
add third option for setting workspace names

rk1a committed:
fixes tabs

rk1a committed:
removes unnecessary code in click handler

fthx commented:
Hi, thanks for that.
Could you test your version for some time and check logs ("fthx", "js") to see if there is any error/warning?
Right now I don't have the time to review all that, but fast renaming is in the scope of this extension as it's used by a lot of people.

rk1a commented:
> Hi, thanks for that. Could you test your version for some time and check logs ("fthx", "js") to see if there is any error/warning? Right now I don't have the time to review all that, but fast renaming is in the scope of this extension as it's used by a lot of people.

Hi, sure, checking logs while clicking the bar always gives me an error which is however unrelated to the code I added:
```
JS ERROR: TypeError: dockManager is null
getRunningApps@/usr/share/gnome-shell/extensions/ubuntu-dock@ubuntu.com/locations.js:653:9
wrapWindowsManagerApp/fileManagerApp._updateWindows@/usr/share/gnome-shell/extensions/ubuntu-dock@ubuntu.com/locations.js:258:9
wrapWindowsManagerApp/windowsChangedId<@/usr/share/gnome-shell/extensions/ubuntu-dock@ubuntu.com/locations.js:250:24
_toggle_ws@/home/me/.local/share/gnome-shell/extensions/workspaces-bar@fthx/extension.js:124:62
_update_ws/<@/home/me/.local/share/gnome-shell/extensions/workspaces-bar@fthx/extension.js:114:59
```
With my version I see this log message times the number of workspaces when first right-clicking on the bar and when I rename the workspaces:
```
clutter_input_focus_set_cursor_location: assertion 'clutter_input_focus_is_focused (focus)' failed
```
But there is no indication from where..

btw, I'm on gnome 40.4.0

rk1a committed:
do menu building on ws number changed

rk1a committed:
Fixes for changeing num ws, whitespace handling
- ws name array is updated according to num ws
- when renaming whitespaces are trimmed
- only small names are padded in ws bar

fthx commented:
I will look at this soon, during holidays.

muhammad-saleh commented:
@fthx Did you have time to look into this? I think it will be a great add.
